sdk/storage/Azure.Storage.Common.DataMovement/tests/FileSystemScannerTests.cs

```diff
@@ -146,11 +145,13 @@ public void ScanMultiplePaths()
 
             string file = CreateRandomFile(_temp);
 
-            FilesystemScanner scanFolder = new FilesystemScanner(folder);
-            FilesystemScanner scanFile = new FilesystemScanner(file);
+            FileSystemScannerFactory folderFactory = new FileSystemScannerFactory(folder);
+            FileSystemScanner scanFolder = folderFactory.BuildFilesystemScanner();
+            FileSystemScannerFactory fileFactory = new FileSystemScannerFactory(file);
+            FileSystemScanner scanFile = fileFactory.BuildFilesystemScanner();
```

Comment:
do we have any test that covers sunny day scenario with nested folders ?

Reply:
The first test sets up a folder with two direct children files (one is unreadable), and two subfolders, one with and one without read permissions. The tests succeeds with expected output being the two direct children and the child inside the readable subfolder, so that covers both nested folder cases. Should this be split up from one test?